loadtest/loadtest.go

```diff
@@ -128,18 +149,34 @@ func (lt *LoadTester) removeUsers(numUsers int) (int, error) {
 	}
 
 	var wg sync.WaitGroup
+
 	wg.Add(numUsers)
 	// TODO: Add a way to make how a user is removed decidable from the upper layer (the user of this API),
 	// for example by passing a typed constant (e.g. random, first, last).
-	for i := 0; i < numUsers; i++ {
-		go func(i int) {
+	var stopped []int
+
+	for id := range lt.activeControllers {
+		if len(stopped) == numUsers {
+			break
+		}
+
+		go func(id int) {
 			defer wg.Done()
-			lt.controllers[activeUsers-1-i].Stop()
-		}(i)
+			if c, ok := lt.activeControllers[id]; ok {
+				c.Stop()
+			}
+		}(id)
+
+		stopped = append(stopped, id)
 	}
+
 	wg.Wait()
 
-	lt.controllers = lt.controllers[:activeUsers-numUsers]
+	for _, id := range stopped {
+		lt.idleControllers[id] = lt.activeControllers[id]
```

Comment:
Same as above. Maybe check whether we are overwriting an existing idle controller.